actionpack/test/dispatch/request/json_params_parsing_test.rb

```diff
@@ -38,10 +38,15 @@ def teardown
   end
 
   test "does not parse unregistered media types such as application/vnd.api+json" do
-    assert_parses(
-      {},
-      "{\"person\": {\"name\": \"David\"}}", "CONTENT_TYPE" => "application/vnd.api+json"
+    output = StringIO.new
+    refute_parses(
+      :unsupported_media_type,
+      "{\"person\": {\"name\": \"David\"}}",
+      "CONTENT_TYPE" => "application/vnd.api+json", "action_dispatch.logger" => ActiveSupport::Logger.new(output)
     )
+    output.rewind && err = output.read
+    assert err =~ /ActionController::UnsupportedMediaType \(#{Regexp.escape("application/vnd.api+json")}\):/
+    assert_equal("", @response.body)
```

Comment:
No need to assert response.body here.  I'm ok with removing it. It appears to be tested in actionpack/test/dispatch/show_exceptions_test.rb